decart_sdk/errors.py

```diff
@@ -0,0 +1,60 @@
+from typing import Any, Optional
+
+
+class DecartSDKError(Exception):
+    def __init__(
+        self,
+        code: str,
+        message: str,
+        data: Optional[dict[str, Any]] = None,
+        cause: Optional[Exception] = None,
+    ) -> None:
+        self.code = code
+        self.message = message
+        self.data = data or {}
+        self.cause = cause
+        super().__init__(message)
+
+    def __str__(self) -> str:
+        return f"[{self.code}] {self.message}"
+
+    def __repr__(self) -> str:
+        return f"DecartSDKError(code={self.code!r}, message={self.message!r})"
+
+
+class ErrorCodes:
```

Comment:
Hmm I think it's more common to use subclasses than error codes. (And I personally am lazy and usually have maximum one error class (if any) and just play with the message string :) )